images/ipfailover/keepalived/lib/failover-functions.sh

```diff
@@ -12,6 +12,29 @@ readonly KEEPALIVED_CONFIG="/etc/keepalived/keepalived.conf"
 readonly KEEPALIVED_DEFAULTS="/etc/sysconfig/keepalived"
 
 
+function cleanup() {
+  echo "  - Cleaning up ... "
+  [ -n "$1" ] && kill -TERM $1
+
+  local interface=$(get_network_device "$NETWORK_INTERFACE")
+  local vips=$(expand_ip_ranges "$HA_VIPS")
+  echo "  - Releasing VIPs ${vips} (interface ${interface}) ... "
+
+  local regex='^.*?/[0-9]+$'
+
+  for vip in ${vips}; do
```

Comment:
Not in this specific case if the 2 functions fail. They do return defaults as of now ... still the future of all those {jedi,functions} uncertain is!!